packages/runtime/src/next/tests/client.get-component-snapshot.test.ts

```diff
@@ -37,63 +37,73 @@ describe('getComponentSnapshot using v1 element tree endpoint', () => {
     expect(result.document.data).toBeNull()
   })
 
-  test('successfully performs locale fallback by requesting base locale tree after receiving a 404 response for a locale variant tree', async () => {
-    // Arrange
-    const client = createTestClient()
-    const localeToTest = 'fr-FR'
-    const baseLocale = null
-
-    // mock base locale tree document
-    const treeId = 'myTree123'
-    const elementTreeName = 'myElementTree'
-    const elementTreeKey = 'abc123'
-    const document: MakeswiftComponentDocument = {
-      id: treeId,
-      name: elementTreeName,
-      data: {
-        type: 'myType',
-        key: elementTreeKey,
-        props: {},
-      },
-      locale: baseLocale,
-      siteId: 'mySiteId',
-      inheritsFromParent: false,
-    }
-
-    /*
+  test.each([
+    { treeId: 'myTree123', locale: 'fr-FR' },
+    { treeId: 'unsafe:url;chars=@&?❔🤷', locale: 'fr-FR' },
+    { treeId: '/blog/slug', locale: 'en-US' },
+  ])(
```

Comment:
Expanded tests to probe the API with "problematic" IDs (review with the "Hide whitespace" toggle on to see the actual diffs).

Reply:
<img width="1715" alt="passing tests" src="https://github.com/user-attachments/assets/247caa8d-0b6f-4231-abe8-b744efa6fc97" />